_inc/connect-button.js

```diff
@@ -80,12 +80,20 @@ jQuery( document ).ready( function( $ ) {
 		},
 		receiveData: function( event ) {
 			if (
-				event.origin === jpConnect.jetpackApiDomain &&
-				event.source === jetpackConnectIframe.get( 0 ).contentWindow &&
-				event.data === 'close'
+				event.origin !== jpConnect.jetpackApiDomain ||
+				event.source !== jetpackConnectIframe.get( 0 ).contentWindow
```

Comment:
Can we simplify this even more. 
```
if ( event.origin !== jpConnect.jetpackApiDomain ) { 
return;
}
if ( event.source !== jetpackConnectIframe.get( 0 ).contentWindow ) {
return;
}
```

That would make it easier to grasp the conditions that need to get meet.